### PR TITLE
Packages/humagne/meep

### PR DESCRIPTION
**Pull request by nrichart**
Updating meep and NanoComp softwares new versions

https://github.com/epfl-scitas/spack-packagelist/pull/189

### CODE REVIEW

**Comment by rmsds** [comment]
Closing and reopening in an attempt to re-trigger Travis.

File: var/spack/repos/builtin/packages/meep/package.py [hunk 1]
```diff
@@ -72,6 +86,11 @@ def configure_args(self):
         else:
             config_args.append('--without-hdf5')
 
+        # make meep-python fail to compile
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/meep/package.py:93]: This comment is not very clear to me. Is the `--with-scheme` implicit and it requires `--enable-maintainer-mode` for the `make meep-python` to work?

For an upstream PR we could make trigger this choice by a variant.

**Review by rmsds — COMMENTED** [review]
In general it looks good to me (but I didn't test the build). 

The failing tests are on unrelated coverage tests.

File: var/spack/repos/builtin/packages/meep/package.py (hunk 1 above)
Comment (nrichart) [reply var/spack/repos/builtin/packages/meep/package.py:93]: `--with-scheme` is on by default and fails to compile if `--enable-maintainer-mode` is not set.
If I do `--without-scheme` it fails to compile the python binding
I did not try `--whithout-scheme --whithout-python`

File: var/spack/repos/builtin/packages/meep/package.py [hunk 2]
```diff
@@ -72,6 +86,11 @@ def configure_args(self):
...
+        # config_args.append('--without-scheme')
+        # --with-scheme need maintainer mode...
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/meep/package.py:95]: Maybe something like this:
```suggestion
        # meep-python + scheme (implicit) fails to build without maintainer mode
```

Comment (rmsds) [reply var/spack/repos/builtin/packages/meep/package.py:95]: Also, aren't we maybe missing a dependency on `python`?

Comment (nrichart) [reply var/spack/repos/builtin/packages/meep/package.py:95]: Yes that is correct, I added the missing dependencies in 59546ee

**Pull request merged**